sdk/identity/identity/src/util/validateMultiTenant.ts

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { GetTokenOptions } from "@azure/core-auth";
+
+/**
+ * @internal
+ */
+export const multiTenantError = new Error(
+  "Multi-tenant authentication was attempted, but multi-tenant authentication was not enabled in this credential instance."
+);
+
+/**
+ * Verifies whether locally assigned tenants are equal to tenants received through getToken.
+ * @internal
+ */
+export function validateMultiTenantRequest(
+  allowMultiTenantAuthentication?: boolean,
+  tenantId?: string,
+  getTokenOptions?: GetTokenOptions
+): void {
+  if (
+    !allowMultiTenantAuthentication &&
+    getTokenOptions?.tenantId &&
+    tenantId &&
+    getTokenOptions.tenantId !== tenantId
+  ) {
+    throw multiTenantError;
```

Comment:
nit: indenting seems off here

Reply:
Our indenting is automated 🤔 I’ll try to run it again to see if it makes any difference.

Reply:
This is how it gets generated 🤔 confirmed